### DISCUSSION TITLE
Ensure empty array for a manyOf datatype

### DISCUSSION THREAD

**Discussion by tarjei** [mswjs/data#220] [Q&A]
Hi, I got an entity with 

```js
{
  module: {
    lessons: manyOf('lesson')
  }
}
```

How should I write this so that lessons are always an array i.e never undefined?

**Comment by kettanaito** [comment]
Hey, @tarjei. Thanks for asking this. 

Your model definition seems correct. Here's a recap below:

```js
import { factory, primaryKey, manyOf } from '@mswjs/data'

const db = factory({
  module: {
    id: primaryKey(String),
    lessons: manyOf('lesson')
  },
  lesson: {
    id: primaryKey(String)
  }
})

db.module.create({
  lessons: [
    db.lesson.create({ id: '1' }),
    db.lesson.create({ id: '2' }),
    db.lesson.create({ id: '3' }),
  ]
})
```

1. `module.lessons` is _always_ an array. It's either an empty array if you haven't provided any values, or an array with lessons like in the example above. This is the default behavior of `manyOf`. 
2. You cannot update a non-nullable relationship to `null`/`undefined`. If you trying doing so, you will get an error:
```js
db.module.update({
  where: { id: { equals: '1' } },
  data: { lessons: undefined }
})

// Error: Failed to update a "MANY_OF" relationship to "lesson
// at "module.lessons" (id: "1"): cannot update a non-nullable relationship to null.
```

You can also have nullable relationships by wrapping any relationship in `nullable`. See README for more details.
